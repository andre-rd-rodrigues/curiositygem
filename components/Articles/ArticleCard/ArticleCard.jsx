import { fadeInVariant, motion } from "assets/motion/motionVariants";
import AppImage from "components/AppImage/AppImage";
import Link from "next/link";
import { convertDate } from "utils/helpers/date";
import styles from "./articlecard.module.scss";

const ArticleCard = ({ article }) => {
  return (
    <motion.div variants={fadeInVariant} className={styles.container}>
      <div className={styles.wrapper}>
        <Link href={`/article/${article.slug}`} passHref>
          <AppImage src={article.coverPhoto.url} className={styles.image} />
          <div className={styles.subtitle}>
            <p>{article.category}</p>
            <p>{convertDate(article.createdAt)}</p>
          </div>
          <div className={styles.body}>
            <h5>{article.title}</h5>
            <p>{article.description}</p>
          </div>
        </Link>
        <div className={styles.footer}>
          <Link
            className={styles.readMore}
            href={`/article/${article.slug}`}
            onClick={() => (scrollTop ? window.scrollTo(0, 0) : null)}
          >
            Read more
          </Link>
        </div>
      </div>
    </motion.div>
  );
};

export default ArticleCard;
