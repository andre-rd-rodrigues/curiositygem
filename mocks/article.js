const articleMock = {
  slug: "transform-your-home-into-a-smart-home-5-ways-to-automate-your-house",
  title: "Transform Your Home into a Smart Home: 5 Ways to Automate Your House",
  description:
    "Are you tired of constantly forgetting to turn off the lights or adjust the thermostat when you leave the house? It might be time to consider automating your home. In this blog post, we&apos;ll give you five ways to transform your home into a smart home, including using smart thermostats, smart lighting, smart appliances, smart security systems, and smart home hubs. Automating your home can make your life easier, save you energy and money, and give you peace of mind. Learn how to get started with home automation today.",
  category: "technology",
  createdAt: "11/25/2022",
  isTopArticle: false,
  coverPhoto: {
    url: "https://images.unsplash.com/photo-1558002038-1055907df827?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2940&q=80"
  },
  content: {
    html: "<div><h2>Take control of your home</h2><p>Are you tired of constantly forgetting to turn off the lights or adjust the thermostat when you leave the house? It might be time to consider automating your home.<br /> Home automation refers to the use of technology to control and manage the various systems and devices in your home, such as lighting, heating and cooling, and appliances. Not only can home automation make your life easier and more convenient, it can also help you save energy and money. Here are 5 ways to automate your home!</p>GOOGLE_AD_HERE</div>"
  },
  writer: "Andre Rodrigues"
};

export default articleMock;
