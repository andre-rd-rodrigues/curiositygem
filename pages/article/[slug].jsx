import AppIcon from "components/AppIcon/AppIcon";
import AppImage from "components/AppImage/AppImage";
import Loading from "components/Loading/Loading";
import PageContainer from "components/PageContainer/PageContainer";
import ShareLinks from "components/ShareLinks/ShareLinks";
import { RouterContext } from "context/route-context";
import { NextSeo } from "next-seo";
import Link from "next/link";
import { ARTICLE_QUERY, graphcms, SLUGLIST } from "pages/api/graphQL/main";
import { useContext, useEffect } from "react";
import parser from "react-html-parser";
import styles from "styles/articlepage.module.scss";
import { convertDate } from "utils";
import { getPreviousRoute } from "utils/helpers/navigation";

function Article({ post: article }) {
  const { previousRoute } = useContext(RouterContext);

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  return (
    <>
      <NextSeo
        title={article.title}
        description={article.description}
        canonical={`https://www.curiositygem.com/article/${article.slug}`}
        openGraph={{
          url: `https://www.curiositygem.com/article/${article.slug}`,
          title: article.title,
          description: article.description,
          datePublished: article.publishedDate,
          authorName: "AndrÃ© Rodrigues",
          images: [
            {
              url: article.coverPhoto.url,
              alt: `${article.title} article`
            }
          ]
        }}
      />

      <PageContainer>
        <div className={styles.container}>
          <Link href={getPreviousRoute(previousRoute)}>
            <AppIcon
              icon="arrow-left"
              size={30}
              color="grey"
              className={styles.arrowBack}
            />
          </Link>
          <div>
            {article ? (
              <>
                <div>
                  <h1 className={styles.title}>{article.title}</h1>
                  <div className={styles.header}>
                    <div className={styles.subtitle}>
                      <p>{convertDate(article.publishedDate)}</p> <span>|</span>{" "}
                      <p>{article.category}</p>
                    </div>
                    <ShareLinks
                      slug={article.slug}
                      image={article.coverPhoto.url}
                      description={article.description}
                      className={styles.socialMedia}
                    />
                  </div>
                </div>
                <AppImage
                  className={styles.headerImage}
                  src={article.coverPhoto.url}
                />
                <div className={styles.content}>
                  {parser(article.content.html)}
                </div>
              </>
            ) : (
              <Loading />
            )}
          </div>
        </div>
      </PageContainer>
    </>
  );
}

export default Article;

export async function getStaticPaths() {
  const { posts } = await graphcms.request(SLUGLIST);

  const paths = posts.map((post) => {
    return { params: { slug: post.slug } };
  });

  return {
    paths,
    fallback: false
  };
}

export async function getStaticProps({ params }) {
  const slug = params.slug;

  const data = await graphcms.request(ARTICLE_QUERY, { slug });

  const post = data.post;

  return {
    props: { post: post },
    revalidate: 10
  };
}
