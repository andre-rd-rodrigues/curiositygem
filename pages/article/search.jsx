import Head from "components/AppHead/AppHead";
import ArticlesGrid from "components/Articles/ArticlesGrid/ArticlesGrid";
import Loading from "components/Loading/Loading";
import NoResults from "components/NoResults/NoResults";
import PageContainer from "components/PageContainer/PageContainer";
import { ArticlesContext } from "context/articles-context";
import useArticles from "hooks/useArticles";
import useRouteQuery from "hooks/useRouteQuery";
import { useContext, useEffect, useState } from "react";
import styles from "styles/resultspage.module.scss";

const ResultsPage = () => {
  const [articlesMatched, setArticlesMatched] = useState([]);

  const { articles, isError } = useContext(ArticlesContext);

  const { getArticlesByInput, getArticlesByCategory } = useArticles(articles);

  const { input, category, router } = useRouteQuery();

  const searchedValue = input || category;

  const renderContent = () => {
    if (!articles) {
      return <Loading />;
    }

    if (isError || !articlesMatched.length) {
      return <NoResults searchValue={searchedValue} />;
    }

    return <ArticlesGrid articles={articlesMatched} />;
  };

  //Lifecycle
  useEffect(() => {
    if (!articles) {
      return;
    }

    let articlesSearched;

    if (input) {
      articlesSearched = getArticlesByInput(input);
    }
    if (category) {
      articlesSearched = getArticlesByCategory(category);
    }

    setArticlesMatched(articlesSearched || []);
  }, [router.asPath, input, category, articles]);

  return (
    <>
      <Head
        description="Curiosity Gem is a blog that shares hidden gems of knowledge on a wide range of topics. From personal development to technology and science, this blog is dedicated to providing you with informative, engaging, and thought-provoking content. Join us on a journey of exploration and discovery and find the knowledge you've been seeking. We're excited to share this latest findings with you and explore new ideas together!"
        title="Curiosity Gem &bull; Discover hidden gems of knowledge"
      />
      <PageContainer
        color="dark"
        bgColor="grey-yellow"
        className={styles.container}
      >
        <div className={styles.results}>
          <div className={styles.title}>
            <h1>Results for:</h1>
            <h2>{searchedValue}</h2>
          </div>
          {renderContent()}
        </div>
      </PageContainer>
    </>
  );
};

export default ResultsPage;
