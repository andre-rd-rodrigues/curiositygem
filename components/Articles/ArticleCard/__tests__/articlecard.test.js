import { render, screen } from "@testing-library/react";
import ArticleCard from "../ArticleCard";
import "@testing-library/jest-dom";
import article from "mocks/article";

describe("Article Card", () => {
  it("renders component unchanged", () => {
    const { container } = render(<ArticleCard article={article} />);

    expect(container).toMatchSnapshot();
  });

  it("renders article from props correctly", () => {
    render(<ArticleCard article={article} />);

    const articleTitleElement = screen.getByText(
      /Transform Your Home into a Smart Home: 5 Ways to Automate Your House/i
    );
    const articleDescriptionElement = screen.getByText(
      /Are you tired of constantly forgetting to turn off the lights or adjust the thermostat when you leave the house?/i
    );

    expect(articleTitleElement).toBeInTheDocument();
    expect(articleDescriptionElement).toBeInTheDocument();
  });
});
