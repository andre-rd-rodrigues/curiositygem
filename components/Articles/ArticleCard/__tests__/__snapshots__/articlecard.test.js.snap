// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Card renders component unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="wrapper"
    >
      <a
        href="/article/transform-your-home-into-a-smart-home-5-ways-to-automate-your-house"
      >
        <div
          class="container image"
          style="background-image: url(https://images.unsplash.com/photo-1558002038-1055907df827?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2940&q=80); height: 30vw;"
        />
        <div
          class="subtitle"
        >
          <p>
            technology
          </p>
          <p />
        </div>
        <div
          class="body"
        >
          <h5>
            Transform Your Home into a Smart Home: 5 Ways to Automate Your House
          </h5>
          <p>
            Are you tired of constantly forgetting to turn off the lights or adjust the thermostat when you leave the house? It might be time to consider automating your home. In this blog post, we&apos;ll give you five ways to transform your home into a smart home, including using smart thermostats, smart lighting, smart appliances, smart security systems, and smart home hubs. Automating your home can make your life easier, save you energy and money, and give you peace of mind. Learn how to get started with home automation today.
          </p>
        </div>
      </a>
      <div
        class="footer"
      >
        <a
          class="readMore"
          href="/article/transform-your-home-into-a-smart-home-5-ways-to-automate-your-house"
        >
          Read more
        </a>
      </div>
    </div>
  </div>
</div>
`;
